crates/dt-workspace/src/lib.rs

```diff
@@ -0,0 +1,58 @@
+use std::{
+    borrow::Cow,
+    path::{Path, PathBuf},
+};
+
+use config::{Config, ConfigError};
+use thiserror::Error;
+
+pub mod config;
+mod search;
+
+#[derive(Debug)]
+pub struct Workspace {
+    path: PathBuf,
+
+    config: Config,
+}
+
+impl Workspace {
+    pub fn try_new(override_path: Cow<'static, Path>) -> Result<Self, WorkspaceError> {
+        let mut path = override_path.to_path_buf();
+
+        if !path.is_dir() {
+            path = PathBuf::from(".");
+        }
+
+        let config = match Config::load_from_parent(&path) {
+            Ok(config) => config,
+            Err(_) => Config::default(),
+        };
```

Comment:
```suggestion
        let config = match Config::load_from_parent(&path) {
            Err(ConfigError::Fs(err)) if err.kind() == std::io::ErrorKind::NotFound => Config::default(),
            other => other?
        };
```
Handle errors other than `NotFound`.